[quantization] Move quantized::add_scalar to ATen

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27258 [quantization] Move quantized::mul_scalar to ATen
* **#27257 [quantization] Move quantized::add_scalar to ATen**

Differential Revision: [D17731566](https://our.internmc.facebook.com/intern/diff/D17731566)

Commit:
[quantization] Move quantized::add_scalar to ATen

Commit:
Update on "[quantization] Move quantized::add_scalar to ATen"

Commit:
Update on "[quantization] Move quantized::add_scalar to ATen"


Differential Revision: [D17731566](https://our.internmc.facebook.com/intern/diff/D17731566)

dzhulgakov commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -254,18 +255,36 @@
+
+- func: add_relu.Tensor(Tensor self, Tensor other) -> Tensor
+  use_c10_dispatcher: full
+  variants: function, method
```
don't make it a function and put into `nn` namespace

dzhulgakov commented on torch/nn/quantized/modules/functional_modules.py:
```diff
@@ -137,7 +137,7 @@ def add(self, x, y):
     r"""Operation equivalent to ``torch.ops.quantized.add(Tensor, float)``"""
     def add_scalar(self, x, y):
```
I wonder whether we should drop it and just make a dispatch on "is scalar" inside the op. The difference is that the arguments would be different (passing or not qscale).

raghuramank100 commented:
Similar comments to add_mul PR, otherwise looks good

z-a-f reviewed (approved):
LGTM

Commit:
Update on "[quantization] Move quantized::add_scalar to ATen"


Differential Revision: [D17731566](https://our.internmc.facebook.com/intern/diff/D17731566)
